ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_linalg.py

```diff
@@ -528,6 +528,87 @@ def test_paddle_solve(
     )
 
 
+# cholesky_solve
+@st.composite
+def _get_cholesky_matrix(draw):
+    # batch_shape, random_size, shared
+    input_dtype = draw(
+        st.shared(
+            st.sampled_from(draw(helpers.get_dtypes("float"))),
+            key="shared_dtype",
+        )
+    )
+    shared_size = draw(
+        st.shared(helpers.ints(min_value=2, max_value=4), key="shared_size")
+    )
+    gen = draw(
+        helpers.array_values(
+            dtype=input_dtype,
+            shape=tuple([shared_size, shared_size]),
+            min_value=2,
+            max_value=5,
+        ).filter(lambda x: np.linalg.cond(x.tolist()) < 1 / sys.float_info.epsilon)
+    )
+    spd = np.matmul(gen.T, gen) + np.identity(gen.shape[0])
+    spd_chol = np.linalg.cholesky(spd)
+    probability = draw(st.floats(min_value=0, max_value=1))
+    if probability > 0.5:
+        spd_chol = spd_chol.T  # randomly transpose the matrix
+    return input_dtype, spd_chol
+
+
+@st.composite
+def _get_second_matrix(draw):
+    # batch_shape, shared, random_size
+    input_dtype = draw(
+        st.shared(
+            st.sampled_from(draw(helpers.get_dtypes("float"))),
+            key="shared_dtype",
+        )
+    )
+    shared_size = draw(
+        st.shared(helpers.ints(min_value=2, max_value=4), key="shared_size")
+    )
+    return input_dtype, draw(
+        helpers.array_values(
+            dtype=input_dtype, shape=tuple([shared_size, 1]), min_value=2, max_value=5
+        )
+    )
```

Comment:
You don't need to copy-paste other tests' helpers. It's cleaner to just import them from the file you found them in.

Reply:
Actually I tried that but that function was expecting a tensor. And the tests didn't pass. Yet still i would give a try again